File: source/aoBlogs2/Controllers/genericController.vb
﻿Option Explicit On
Option Strict On

Imports System.Linq
Imports Contensive.Addons.Blog.Models
Imports Contensive.Addons.Blog.Views
Imports Contensive.BaseClasses

Namespace Controllers
    Public NotInheritable Class genericController
        Private Sub New()
        End Sub
        '
        '====================================================================================================
        ''' <summary>
        ''' if date is invalid, set to minValue
        ''' </summary>
        ''' <param name="srcDate"></param>
        ''' <returns></returns>
        Public Shared Function encodeMinDate(srcDate As DateTime) As DateTime
            Dim returnDate As DateTime = srcDate
            If srcDate < New DateTime(1900, 1, 1) Then
                returnDate = DateTime.MinValue
            End If
            Return returnDate
        End Function
        '
        '====================================================================================================
        ''' <summary>
        ''' if valid date, return the short date, else return blank string 
        ''' </summary>
        ''' <param name="srcDate"></param>
        ''' <returns></returns>
        Public Shared Function getShortDateString(srcDate As DateTime) As String
            Dim returnString As String = ""
            Dim workingDate As DateTime = encodeMinDate(srcDate)
            If Not isDateEmpty(srcDate) Then
                returnString = workingDate.ToShortDateString()
            End If
            Return returnString
        End Function
        '
        '====================================================================================================
        '
        Public Shared Function isDateEmpty(srcDate As DateTime) As Boolean
            Return (srcDate < New DateTime(1900, 1, 1))
        End Function
        '
        '====================================================================================================
        '
        Public Shared Function getSortOrderFromInteger(id As Integer) As String
            Return id.ToString().PadLeft(7, "0"c)
        End Function
        '
        '====================================================================================================
        '
        Public Shared Function getDateForHtmlInput(source As DateTime) As String
            If isDateEmpty(source) Then
                Return ""
            Else
                Return source.Year.ToString() + "-" + source.Month.ToString().PadLeft(2, "0"c) + "-" + source.Day.ToString().PadLeft(2, "0"c)
            End If
        End Function
        '
        '====================================================================================================
        '
        Public Shared Function convertToDosPath(sourcePath As String) As String
            Return sourcePath.Replace("/", "\")
        End Function
        '
        '====================================================================================================
        '
        Public Shared Function convertToUnixPath(sourcePath As String) As String
            Return sourcePath.Replace("\", "/")
        End Function
        '
        '====================================================================================================
        ''' <summary>
        ''' is a member of a group in the group model list
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <param name="groupList"></param>
        ''' <returns></returns>
        Public Shared Function isGroupListMember(cp As CPBaseClass, groupList As List(Of GroupModel)) As Boolean
            For Each Group In groupList
                If cp.User.IsInGroup(Group.name) Then Return True
            Next
            Return False
        End Function
        '
        '====================================================================================================
        ''' <summary>
        ''' Return a shortened version of the copy
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <param name="rawCopy"></param>
        ''' <param name="MaxLength"></param>
        ''' <returns></returns>
        Friend Shared Function getBriefCopy(cp As CPBaseClass, rawCopy As String, MaxLength As Integer) As String
            Try
                Dim Copy As String = cp.Utils.ConvertHTML2Text(rawCopy)
                If Len(Copy) > MaxLength Then
                    Copy = Left(Copy, MaxLength)
                    Copy = Copy & "..."
                End If
                Return Copy
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
                Return ""
            End Try
        End Function
        '
        '====================================================================================
        '
        Public Shared Function getFormTableRow2(cp As CPBaseClass, Innards As String) As String
            Return "<tr>" _
                & "<td colspan=2 width=""100%"">" & Innards & "</td>" _
                & "</tr>"
        End Function
        '
        '====================================================================================
        '
        Public Shared Function getFormTableRow(cp As CPBaseClass, FieldCaption As String, Innards As String, Optional AlignLeft As Boolean = True) As String
            '
            Dim Stream As String = ""
            Try
                Dim AlignmentString As String
                '
                If Not AlignLeft Then
                    AlignmentString = " align=right"
                Else
                    AlignmentString = " align=left"
                End If
                '
                Stream = Stream & "<tr>"
                Stream = Stream & "<td Class=""aoBlogTableRowCellLeft"" " & AlignmentString & ">" & FieldCaption & "</td>"
                Stream = Stream & "<td Class=""aoBlogTableRowCellRight"">" & Innards & "</td>"
                Stream = Stream & "</tr>"
                '
                ' GetFormTableRow = Stream
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
            Return Stream
        End Function
        '
        '====================================================================================
        '
        Public Shared Function getField(cp As CPBaseClass, RequestName As String, Height As Integer, Width As Integer, MaxLenghth As Integer, DefaultValue As String) As String
            Dim result As String = ""
            Try
                If Height = 0 Then
                    Height = 1
                End If
                If Width = 0 Then
                    Width = 25
                End If
                '           
                result = cp.Html.InputText(RequestName, DefaultValue, 255)
                result = Replace(result, "<INPUT ", "<INPUT maxlength=""" & MaxLenghth & """ ", 1, 99, CompareMethod.Text)
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
            Return result
        End Function
        '
        Public Shared Function addEditWrapper(ByVal cp As CPBaseClass, ByVal innerHtml As String, ByVal recordId As Integer, ByVal recordName As String, ByVal contentName As String) As String
            If (Not cp.User.IsEditingAnything) Then Return innerHtml
            Dim header As String = cp.Content.GetEditLink(contentName, recordId.ToString(), False, recordName, True)
            Dim content As String = cp.Html.div(innerHtml, "", "")
            Return cp.Html.div(header + content, "", "ccEditWrapper")
        End Function

    End Class
End Namespace

File: source/aoBlogs2/Controllers/genericController.cs
using System;
using System.Collections.Generic;
using Contensive.Addons.Blog.Models;
using Contensive.BaseClasses;
using Contensive.Models.Db;
using Microsoft.VisualBasic;

namespace Contensive.Addons.Blog.Controllers {
    public sealed class genericController {
        private genericController() {
        }
        // 
        // ====================================================================================================
        /// <summary>
        /// if date is invalid, set to minValue
        /// </summary>
        /// <param name="srcDate"></param>
        /// <returns></returns>
        public static DateTime encodeMinDate(DateTime srcDate) {
            var returnDate = srcDate;
            if (srcDate < new DateTime(1900, 1, 1)) {
                returnDate = DateTime.MinValue;
            }
            return returnDate;
        }
        // 
        // ====================================================================================================
        /// <summary>
        /// if valid date, return the short date, else return blank string 
        /// </summary>
        /// <param name="srcDate"></param>
        /// <returns></returns>
        public static string getShortDateString(DateTime srcDate) {
            string returnString = "";
            var workingDate = encodeMinDate(srcDate);
            if (!isDateEmpty(srcDate)) {
                returnString = workingDate.ToShortDateString();
            }
            return returnString;
        }
        // 
        // ====================================================================================================
        // 
        public static bool isDateEmpty(DateTime srcDate) {
            return srcDate < new DateTime(1900, 1, 1);
        }
        // 
        // ====================================================================================================
        // 
        public static string getSortOrderFromInteger(int id) {
            return id.ToString().PadLeft(7, '0');
        }
        // 
        // ====================================================================================================
        // 
        public static string getDateForHtmlInput(DateTime source) {
            if (isDateEmpty(source)) {
                return "";
            } else {
                return source.Year.ToString() + "-" + source.Month.ToString().PadLeft(2, '0') + "-" + source.Day.ToString().PadLeft(2, '0');
            }
        }
        // 
        // ====================================================================================================
        // 
        public static string convertToDosPath(string sourcePath) {
            return sourcePath.Replace("/", @"\");
        }
        // 
        // ====================================================================================================
        // 
        public static string convertToUnixPath(string sourcePath) {
            return sourcePath.Replace(@"\", "/");
        }
        // 
        // ====================================================================================================
        /// <summary>
        /// is a member of a group in the group model list
        /// </summary>
        /// <param name="cp"></param>
        /// <param name="groupList"></param>
        /// <returns></returns>
        public static bool isGroupListMember(CPBaseClass cp, List<Models.GroupModel> groupList) {
            foreach (var Group in groupList) {
                if (cp.User.IsInGroup(Group.name))
                    return true;
            }
            return false;
        }
        // 
        // ====================================================================================================
        /// <summary>
        /// Return a shortened version of the copy
        /// </summary>
        /// <param name="cp"></param>
        /// <param name="rawCopy"></param>
        /// <param name="MaxLength"></param>
        /// <returns></returns>
        internal static string getBriefCopy(CPBaseClass cp, string rawCopy, int MaxLength) {
            try {
                string Copy = cp.Utils.ConvertHTML2Text(rawCopy);
                if (Strings.Len(Copy) > MaxLength) {
                    Copy = Strings.Left(Copy, MaxLength);
                    Copy = Copy + "...";
                }
                return Copy;
            } catch (Exception ex) {
                cp.Site.ErrorReport(ex);
                return "";
            }
        }
        // 
        // ====================================================================================
        // 
        public static string getFormTableRow2(CPBaseClass cp, string Innards) {
            return "<tr>" + "<td colspan=2 width=\"100%\">" + Innards + "</td>" + "</tr>";

        }
        // 
        // ====================================================================================
        // 
        public static string getFormTableRow(CPBaseClass cp, string FieldCaption, string Innards, bool AlignLeft = true) {
            // 
            string Stream = "";
            try {
                string AlignmentString = AlignLeft ? " align=right" : " align=left";
                Stream = Stream + "<tr>";
                Stream = Stream + "<td Class=\"aoBlogTableRowCellLeft\" " + AlignmentString + ">" + FieldCaption + "</td>";
                Stream = Stream + "<td Class=\"aoBlogTableRowCellRight\">" + Innards + "</td>";
                Stream = Stream + "</tr>";
                return Stream;
            } catch (Exception ex) {
                cp.Site.ErrorReport(ex);
                throw;
            }
        }
        // 
        // ====================================================================================
        // 
        public static string getField(CPBaseClass cp, string RequestName, int Height, int Width, int MaxLenghth, string DefaultValue) {
            string result = "";
            try {
                if (Height == 0) {
                    Height = 1;
                }
                if (Width == 0) {
                    Width = 25;
                }
                // 
                result = cp.Html.InputText(RequestName, DefaultValue, 255);
                result = Strings.Replace(result, "<INPUT ", "<INPUT maxlength=\"" + MaxLenghth + "\" ", 1, 99, CompareMethod.Text);
            } catch (Exception ex) {
                cp.Site.ErrorReport(ex);
            }
            return result;
        }
        // 
        public static string addEditWrapper(CPBaseClass cp, string innerHtml, int recordId, string recordName, string contentName) {

            return cp.Content.GetEditWrapper(innerHtml, contentName, recordId);
        }

    }
}